airflow/hooks/hive_hooks.py

```diff
@@ -121,11 +148,31 @@ def run_cli(self, hql, schema=None, verbose=True, hive_conf=None):
                 for key, value in hive_conf.items():
                     cmd_extra += ['-hiveconf', '{0}={1}'.format(key, value)]
 
-                hive_cmd = [hive_bin, '-f', fname] + cmd_extra
+                hive_cmd = [hive_bin, '-f', fname]
 
                 if self.hive_cli_params:
                     hive_params_list = self.hive_cli_params.split()
-                    hive_cmd.extend(hive_params_list)
+                else:
+                    hive_params_list = []
+
+                if self.mapred_queue:
+                    hive_params_list.append(
```

Comment:
We should probably turn it into an OrderedDict then, because I believe that the ordering matters here. My way makes sure that the queues specified override what's in hive conf and iterating over dictionary elements does not preserve order..